clip_retrieval/clip_inference/open_clip_wrapper.py

```diff
@@ -0,0 +1,25 @@
+"""An autocast wrapper for OpenClip models"""
+
+from torch import autocast, nn, device as torch_device
+
+
+class OpenClipWrapper(nn.Module):
+    """
+    Wrap OpenClip for managing input types
+    """
+
+    def __init__(self, inner_model, device):
+        super().__init__()
+        self.inner_model = inner_model
+        self.device = torch_device(device=device)
+
+    def encode_image(self, image):
+        with autocast(device_type=self.device.type):
```

Comment:
It might be needed to specify float16 by default and/or auto detect if bfloat16 is supported